### PR TITLE
CPLAT-10335 Migrate demo, app and lib components to new boilerplate (rebased version of #479), misc cleanup

### PR DESCRIPTION
**Pull request by greglittlefield-wf**
### Changes

Main changes:
- 7dcc16d- Migrate demo, app and lib components to new boilerplate (#479)
    - Includes the changes from https://github.com/Workiva/over_react/pull/479, but squashed, rebased, and stripped of builder-related changes (which were merged separately)
    - Also converts connect usages to use the new compact syntax
- 68493ab Update ignore comments
    - Using the new boilerplate codemod to move them for factories
- 11711bd48d0a31d5dc66d4f66271712fb812f7f5
Un-inline ErrorBoundaryMixin in legacy RecoverableErrorBoundaryComponent
    - Having the code inline seemed like a mistake made during the refactor; I verified that the code was 100% just the mixin body pasted in, so I put it back so we wouldn't have two versions of that code floating around

Misc:
- 52182f4 Improve assert error message with newly discovered failure case
    - Added in response to @aaronlademann-wf 's testing

Cleanup:
- f69f3e4 Remove leftover function component example
- 8a52923 Export defaultErrorBoundaryLoggerName
    - This was previously exported but got removed in previous commits:

### Testing
QA instructions:
- `example/`, `web/`, and `app/` apps run properly
- Semver report looks good

### CODE REVIEW

**Comment by aviary2-wf** [comment]
## Security Insights
The items listed below may not capture all security relevant changes. Before providing a security review, be sure to review the entire PR for security impact.
<details>
<summary>(7) Security relevant changes were detected</summary>
<li>Watched keyword <code>InnerHtml</code> in <code>lib/src/component/_deprecated/error_boundary.dart</code> line(s) ['111'] added</li>
<li>Watched keyword <code>innerHtml</code> in <code>lib/src/component/_deprecated/error_boundary_recoverable.dart</code> line(s) ['163'] added</li>
<li>Watched keyword <code>InnerHtml</code> in <code>lib/src/component/_deprecated/error_boundary_recoverable.dart</code> line(s) ['190'] added</li>
<li>Watched keyword <code>innerHtml</code> in <code>lib/src/component/error_boundary_recoverable.dart</code> line(s) ['157'] added</li>
<li>Watched keyword <code>InnerHtml</code> in <code>lib/src/component/error_boundary_recoverable.dart</code> line(s) ['184'] added</li>
<li>Watched keyword <code>dangerouslySetInnerHTML</code> in <code>lib/src/component/_deprecated/error_boundary_recoverable.dart</code> line(s) ['191'] added</li>
<li>Watched keyword <code>dangerouslySetInnerHTML</code> in <code>lib/src/component/error_boundary_recoverable.dart</code> line(s) ['185'] added</li>
</details>

## Action Items
* Obtain a [security review](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); reviewer should pay special attention to insights listed above
* Verify [`aviary.yaml`](https://dev.workiva.net/docs/teams/information-security/aviary#configuring-aviary) coverage of security relevant code
---
_Questions or Comments? Reach out on Slack: #support-infosec._

**Comment by semveraudit-wf** [comment]
### Public API Changes

 No changes to the public API found for commit 6fa45324c129bbafd4cb1a18e4b7ec25ce597812

Showing results for 6fa45324c129bbafd4cb1a18e4b7ec25ce597812
> Powered by [semver-audit-service](https://github.com/Workiva/semver-audit-service). Please report any problems by [filing an issue](https://github.com/Workiva/semver-audit-service/issues).
> Reported by the dart semver audit client 2.0.3
[Browse public API](https://semver-audit.wk-dev.wdesk.org/Workiva/over_react/6fa45324c129bbafd4cb1a18e4b7ec25ce597812).

File: lib/src/component/_deprecated/error_boundary_mixins.dart [hunk 1]
```diff
@@ -3,14 +3,13 @@ import 'dart:async';
 import 'package:logging/logging.dart';
 import 'package:meta/meta.dart';
 import 'package:over_react/over_react.dart';
+import 'package:over_react/src/component/error_boundary_api.dart';
 
 part 'error_boundary_mixins.over_react.g.dart';
 
-@visibleForTesting
-const String defaultErrorBoundaryLoggerName = 'over_react.ErrorBoundary';
```
[the remark this answers is not in the document]
Comment (greglittlefield-wf) [reply lib/src/component/_deprecated/error_boundary_mixins.dart:10]: Addressed in https://github.com/Workiva/over_react/pull/482/commits/8a52923b0297565f31160f7bff1c258c202f5bae

File: lib/src/component/error_boundary_recoverable.dart [hunk 2]
```diff
@@ -9,15 +14,224 @@ part 'error_boundary_recoverable.over_react.g.dart';
...
+      for (var i = 0; i < _errorLog.length; i++) {
+        if (_errorLog[i] == errorString && _callStackLog[i].componentStack == info.componentStack) {
+          sameErrorWasThrownTwiceConsecutively = true;
+          break;
+        }
+      }
+
+      if (sameErrorWasThrownTwiceConsecutively) { // [2.1]
+        try { // [2.2.2]
+          _domAtTimeOfError = findDomNode(this)?.innerHtml; // [2.2]
```
Comment (greglittlefield-wf) [inline lib/src/component/error_boundary_recoverable.dart:157]: This is a copy of the code here: https://github.com/Workiva/over_react/blob/8a52923b0297565f31160f7bff1c258c202f5bae/lib/src/component/_deprecated/error_boundary_recoverable.dart#L162

File: lib/src/component/_deprecated/error_boundary_recoverable.dart [hunk 3]
```diff
@@ -0,0 +1,242 @@
...
+      for (var i = 0; i < _errorLog.length; i++) {
+        if (_errorLog[i] == errorString && _callStackLog[i].componentStack == info.componentStack) {
+          sameErrorWasThrownTwiceConsecutively = true;
+          break;
+        }
+      }
+
+      if (sameErrorWasThrownTwiceConsecutively) { // [2.1]
+        try { // [2.2.2]
+          _domAtTimeOfError = findDomNode(this)?.innerHtml; // [2.2]
```
Comment (greglittlefield-wf) [inline lib/src/component/_deprecated/error_boundary_recoverable.dart:162]: This code is a copy of https://github.com/Workiva/over_react/blob/8a52923b0297565f31160f7bff1c258c202f5bae/lib/src/component/_deprecated/error_boundary_mixins.dart#L326

**Comment by kealjones-wk** [comment]
Security +1

- All of these changes have had previous security review approvals, the files are simply moving directories
- Flagged keywords `innerHtml` are not being assigned to from user input.
- One watched term `InnerHtml` was flagged, but it's actually is part of a test ID.

File: lib/src/component/abstract_transition.dart [hunk 4]
```diff
@@ -271,7 +267,7 @@ abstract class AbstractTransitionComponent<T extends AbstractTransitionProps,
 
   @mustCallSuper
   @override
-  void componentDidUpdate(Map prevProps, Map prevState) {
+  void componentDidUpdate(Map prevProps, Map prevState, [dynamic snapshot]) {
```
Comment (joebingham-wk) [inline lib/src/component/abstract_transition.dart:270]: #supernit I know thing were just moved around, but I noticed this is an unused parameter 🤷‍♂ 
```suggestion
  void componentDidUpdate(Map prevProps, Map prevState, [_]) {
```

File: lib/src/component_declaration/component_base.dart [hunk 5]
```diff
@@ -511,9 +511,11 @@ abstract class UiProps extends MapBase
     assert(
         componentFactory != null,
         'componentFactory is null. Possible causes:\n'
-        '1. This is a props map view factory, and should not be invoked.'
-        '2. This is a function component factory that was set up improperly, not wrapping the generated function in `uiFunction`.'
-        '3. componentFactory was erroneously assigned to null on this UiProps instance, potentially in an HOC function.');
+        '1. Something went wrong when initializing the `\$${runtimeType}Factory` variable in the generated code.\n'
+        '   It\'s possible React swallowed errors thrownn during that initialization, so try pausing on caught exceptions to see it.\n'
```
Comment (joebingham-wk) [inline lib/src/component_declaration/component_base.dart:515]: Typo
```suggestion
        '   It\'s possible React swallowed errors thrown during that initialization, so try pausing on caught exceptions to see it.\n'
```

File: web/component1/index.dart [hunk 6]
```diff
@@ -9,25 +10,25 @@ void main() {
...
-    ErrorBoundary()(checkboxToggleButtonDemo()), querySelector('$demoMountNodeSelectorPrefix--checkbox-toggle'));
+    v2.ErrorBoundary()(checkboxToggleButtonDemo()), querySelector('$demoMountNodeSelectorPrefix--checkbox-toggle'));
 
   react_dom.render(
-    ErrorBoundary()(radioToggleButtonDemo()), querySelector('$demoMountNodeSelectorPrefix--radio-toggle'));
+    v2.ErrorBoundary()(radioToggleButtonDemo()), querySelector('$demoMountNodeSelectorPrefix--radio-toggle'));
 
   react_dom.render(
-    (ErrorBoundary()
+    (v2.ErrorBoundary()
```
Comment (joebingham-wk) [inline web/component1/index.dart:31]: #nit not really a big deal at all, but nothing about this example is Component1 anymore, so I think it's funny that it's here haha

File: lib/src/component/error_boundary_api.dart [hunk 7]
```diff
@@ -0,0 +1,17 @@
+import 'package:meta/meta.dart';
```
Comment (sydneyjodon-wk) [inline lib/src/component/error_boundary_api.dart:1]: #nit missing copyright comment 

Also there's a few places where the copyright comment on new files has the wrong year (although maybe you were planning on doing that in a different PR - I can't remember)

Comment (greglittlefield-wf) [reply lib/src/component/error_boundary_api.dart:1]: Yeah, I'm just gonna add/update all the copyrights in a separate PR. Thanks for checking!

**Comment by greglittlefield-wf** [comment]
@joebingham-wk How does merging this as-is and then following up with consumedProps fixes in the PR where I update default consumedProps behavior sound?

**Comment by kealjones-wk** [comment]
Security +1

Refresh

**Pull request merged**